resources/metadata/gradleExecuteBuild.yaml

```diff
@@ -115,6 +115,36 @@ spec:
           - STAGES
           - PARAMETERS
         default: false
+      - name: applyCreateBOMForAllProjects
+        type: bool
+        description: If set to false cycloneDX plugin will be applied in 'rootProject' directive, otherwise 'allprojects' will be directive used
+        scope:
+          - STEPS
+          - STAGES
+          - PARAMETERS
+        default: false
+      - name: applyPublishingForAllProjects
```

Comment:
do we need this flag ? , should we keep it simple that it publish will work for all projects by default . we have another flag for `excludePublishingForProjects` . adding a value to this string should be enough to exclude a project if needed ?

Reply:
it was added mostly for backcompatibility with existing pipelines.
Also, it's adding a small logic here https://github.com/SAP/jenkins-library/blob/4066c4084798da16e63419e74cddbb7c4d1f95ab/cmd/gradleExecuteBuild.go#L51 where instead of using id provided by `artifactPrepareVersion `project's name will be used. I'd like to save this part.